source/translated_table_model.ts

```diff
@@ -28,22 +46,155 @@ export class TranslatedTableModel extends TableModel {
    *                      to this table's columnCount.
    * @throws RangeError - The index specified is not within range.
    */
-  public moveRow(source: number, destination: number): void {}
+  public moveRow(source: number, destination: number): void {
+    if(source > this.rowCount || source < 0) {
```

Comment:
What if ```source === this.rowCount```? Add a unit test for that scenario, ensure it fails, commit your test, then implement a solution, commit your solution. There should be at least one commit showing the failure and then a commit showing it succeed.